src/Symfony/Component/Mailer/Bridge/Amazon/Transport/SesHttpTransport.php

```diff
@@ -11,75 +11,78 @@
 
 namespace Symfony\Component\Mailer\Bridge\Amazon\Transport;
 
+use AsyncAws\Core\Configuration;
+use AsyncAws\Core\Exception\Http\HttpException;
+use AsyncAws\Ses\Input\Destination;
+use AsyncAws\Ses\Input\SendEmailRequest;
+use AsyncAws\Ses\SesClient;
 use Psr\Log\LoggerInterface;
 use Symfony\Component\Mailer\Exception\HttpTransportException;
 use Symfony\Component\Mailer\SentMessage;
-use Symfony\Component\Mailer\Transport\AbstractHttpTransport;
+use Symfony\Component\Mailer\Transport\AbstractTransport;
 use Symfony\Contracts\EventDispatcher\EventDispatcherInterface;
 use Symfony\Contracts\HttpClient\HttpClientInterface;
-use Symfony\Contracts\HttpClient\ResponseInterface;
 
 /**
  * @author Kevin Verschaeve
+ * @author Jérémy Derussé <jeremy@derusse.com>
  */
-class SesHttpTransport extends AbstractHttpTransport
+class SesHttpTransport extends AbstractTransport
 {
-    private const HOST = 'email.%region%.amazonaws.com';
+    /** @var SesClient  */
+    private $sesClient;
 
-    private $accessKey;
-    private $secretKey;
-    private $region;
+    /** @var Configuration */
+    protected $configuration;
 
-    /**
-     * @param string $region Amazon SES region (currently one of us-east-1, us-west-2, or eu-west-1)
-     */
-    public function __construct(string $accessKey, string $secretKey, string $region = null, HttpClientInterface $client = null, EventDispatcherInterface $dispatcher = null, LoggerInterface $logger = null)
+    public function __construct(Configuration $configuration, SesClient $sesClient = null, HttpClientInterface $httpClient = null, EventDispatcherInterface $dispatcher = null, LoggerInterface $logger = null)
```

Comment:
If you pass a SESClient then HttpClient is never used.

Reply:
good catch. I wonder if I should either:
- inject both config and client
- inject only client, but adds public methods 'getConfiguration' in Client

WDYT?

Reply:
I like the idea of only using a SESClient. 

I think it is fair to add `getConfiguration` because `Configuration` is immutable.

Reply:
see https://github.com/async-aws/aws/pull/276